File: java/src/main/java/com/callisto/sdk/resources/Balance.java
package com.callisto.sdk.resources;

import com.callisto.sdk.http.Transport;

import java.util.LinkedHashMap;
import java.util.Map;

/** The balance resource. */
public class Balance {

    private final Transport transport;

    public Balance(Transport transport) {
        this.transport = transport;
    }

    /** Returns the account balance using the default format {@code "full"} and no currency filter. */
    public com.callisto.sdk.models.Balance get() {
        return get("full", null);
    }

    /**
     * Returns the account balance.
     *
     * @param format   response format (defaults to {@code "full"} when {@code null}).
     * @param currency optional currency code filter.
     */
    public com.callisto.sdk.models.Balance get(String format, String currency) {
        Map<String, Object> query = new LinkedHashMap<>();
        query.put("format", format != null ? format : "full");
        query.put("currency", currency);
        return transport.mapper().convertValue(
                transport.request("GET", "/sms/balance", null, query),
                com.callisto.sdk.models.Balance.class);
    }
}

File: ruby/lib/callisto/resources/balance.rb
# frozen_string_literal: true

module Callisto
  # Account balance resource.
  class BalanceResource
    # @param transport [Transport]
    def initialize(transport)
      @t = transport
    end

    # Returns the account balance.
    #
    # @param format [String] response format (default "full")
    # @param currency [String, nil] filter by currency code
    # @return [Balance]
    def get(format: "full", currency: nil)
      data = @t.request("GET", "/sms/balance", query: { "format" => format, "currency" => currency })
      Balance.from_hash(data)
    end
  end
end
